quickwit/quickwit-opentelemetry/src/otlp/span_id.rs

```diff
@@ -42,33 +40,29 @@ impl SpanId {
 
 impl Serialize for SpanId {
     fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
-        let b64span_id = BASE64_STANDARD.encode(self.0);
-        serializer.serialize_str(&b64span_id)
+        let hexspan_id = hex::encode(self.0);
+        serializer.serialize_str(&hexspan_id)
     }
 }
 
 impl<'de> Deserialize<'de> for SpanId {
     fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
     where D: Deserializer<'de> {
-        let b64span_id = String::deserialize(deserializer)?;
+        let hexspan_id = String::deserialize(deserializer)?;
 
-        if b64span_id.len() != SpanId::BASE64_LENGTH {
+        if hexspan_id.len() != SpanId::HEX_LENGTH {
             let message = format!(
-                "base64 span ID must be {} bytes long, got {}",
-                SpanId::BASE64_LENGTH,
-                b64span_id.len()
+                "hex span ID must be {} bytes long, got {}",
+                SpanId::HEX_LENGTH,
+                hexspan_id.len()
             );
             return Err(de::Error::custom(message));
         }
         let mut span_id = [0u8; 8];
-        BASE64_STANDARD
-            // Using the unchecked version here because otherwise the engine gets the wrong size
-            // estimate and fails.
-            .decode_slice_unchecked(b64span_id.as_bytes(), &mut span_id)
-            .map_err(|error| {
-                let message = format!("failed to decode base64 span ID: {:?}", error);
-                de::Error::custom(message)
-            })?;
+        hex::decode_to_slice(hexspan_id, &mut span_id).map_err(|error| {
+            let message = format!("failed to decode hex span ID: {:?}", error);
```

Comment:
```suggestion
            let message = format!("failed to decode hex span ID: {error:?}");
```

This wasn't previously supported. We're trying to convert those opportunistically.